girder_utils/rest.py

```diff
@@ -0,0 +1,22 @@
+from rest_framework.pagination import LimitOffsetPagination
+
+
+class BoundedLimitOffsetPagination(LimitOffsetPagination):
+    """
+    Pagination class that forces pagination with an upper bound.
+
+    DRF's built-in paginators such as LimitOffsetPagination will cause
+    your endpoints to return different structure based on whether or not
+    paging parameters were passed in the request. In most cases it's desirable
+    to force pagination on endpoints that could return large result sets.
+    Using this class has two benefits:
+
+    1. It will always return responses in their paged structure, i.e. a
+       JSON Object with "count", and a "results" Array. This is achieved by
+       providing a default limit that is used if none is passed.
+    2. It enforces a reasonable upper bound on page sizes. Without this, users could
+       circumvent the intent of pagination by simply providing a high enough limit.
+    """
+
+    default_limit = 100
```

Comment:
DRF allows downstreams to globally set `default_limit` [using the `PAGE_SIZE` setting](https://www.django-rest-framework.org/api-guide/pagination/#setting-the-pagination-style).

If we override `paginate_queryset` to never return `None`, it's possible to separate the "always return paginated responses" feature (which I think is the value that we really care about), without dropping support for downstreams to choose their own default `PAGE_SIZE` (or choose not to).

Reply:
It seems simpler to override `get_limit`, OK if I go that route instead?

Reply:
That'd be a different behavior from the pure "always return paginated responses".

Instead, we'd be also effectively force-setting `PAGE_SIZE` for some opinionated (e.g. 100) default value, which has the side effect of always enabling pagination.

For a library like this, I'd personally prefer the more pure, less opinionated behavior whenever possible, but I'd be open to hear what you think.

Reply:
Did you see d8822b8? That approach doesn't override PAGE_SIZE if it's set, just provides a default so that limit is always set to a number.

Reply:
Yeah, see my inline comment to continue to this thread.